### PR TITLE
Don't raise error for probe with unconnected channels

### PR DESCRIPTION
**Pull request by JuliaSprenger**
This PR adds a test case in which an unconnected probe is set for a recording.
Linking an unconnected probe is not a main use case, but might occur e.g. when looping across multiple recording streams and automatically generating probes for all of them.
Previously this case raised an error.

### CODE REVIEW

File: spikeinterface/core/tests/test_baserecording.py [hunk 1]
```diff
@@ -164,6 +164,16 @@ def test_BaseRecording():
...
+    # set unconnected probe
+    probe = Probe(ndim=2)
+    positions = [[0., 0.], [0., 15.], [0, 30.]]
+    probe.set_contacts(positions=positions, shapes='circle',
+                       shape_params={'radius': 5})
+    probe.set_device_channel_indices([-1, -1, -1])
+    probe.create_auto_shape()
+
+    rec_empty_probe = rec.set_probe(probe, group_mode='by_shank')
+
```
Comment (samuelgarcia) [inline spikeinterface/core/tests/test_baserecording.py:176]:
```suggestion
assert rec_empty_probe.channel_ids.size == 0
```

**Comment by samuelgarcia** [comment]
It is a corner case because in that case the recording would have zero channel.
If you need it lets go for it.

I am not sure that the entire spikeinterface would support zero channel size.
But the core should support for it maybe.

@alejoe91 ?

**Review by alejoe91 — APPROVED** [review]
